3723/1/page_printer.rb

```diff
@@ -0,0 +1,36 @@
+require_relative 'sorted_pages'
+
+class PagePrinter
+  attr_reader :rows
+
+  def initialize(rows)
+    @rows = rows
+  end
+
+  def call(option = nil)
+    if option
+      sort_by_number(option) if option.is_a? Integer
+      filter_by_name(option) if option.is_a? String
+    end
+    print_info
+  end
+
+  private
+
+  def sort_by_number(number)
+    @rows = @rows.first(number) if number.positive?
+  end
+
+  def filter_by_name(name)
+    @rows.map do |row|
+      if row.first.include?(name)
+        @rows = []
```

Comment:
You still affect the object you iterate on. Change the spec so it has for example 5 rows and 3 of them match the pattern. I'm pretty sure it will return only one row

Reply:
Now i understand what you mean. Done.